packages/nx/src/plugins/js/lock-file/lock-file.ts

```diff
@@ -191,6 +213,12 @@ export function createLockFile(
       const prunedGraph = pruneProjectGraph(graph, packageJson);
       return stringifyNpmLockfile(prunedGraph, content, normalizedPackageJson);
     }
+    if (packageManager === 'bun') {
+      output.log({
+        title:
+          "Unable to create bun lock files. Run bun install it's just as quick",
```

Comment:
@edbzn  would it be worth calling stringifyYarnLockfile?

Reply:
My thinking is moving the following into a util

```
const lockFile = createLockFile(
      builtPackageJson,
      context.projectGraph,
      packageManager
    );
    writeFileSync(
      `${outDirRelativeToWorkspaceRoot}/${getLockFileName(packageManager)}`,
      lockFile,
      {
        encoding: 'utf-8',
      }
    );
    ```
    
    its duplicated code but run IF check for bun and use the bun CLI to generate the binary file

Reply:
Worth asking @meeroslav, he knows best how to deal with lockfiles

Reply:
Since there doesn't seem to be a way to generate `bun.lockb` without running the install, I would use the `stringifyYarnLockfile` to be as close as possible to the intended package versions.

Reply:
@jaysoo  I've taken another look at this and asked in Bun if we can manually create the lock file but we cant :/ Reading a lock file is absolutely fine packages/nx/src/plugins/js/index.ts. The issue here is Bun lock file is a binary file. Only way to do it would be to execute Bun CLI to create the lock file on bun install

Reply:
Users won't be able to run `bun install --frozen-lockfile` since there would only be `yarn.lock`.

Reply:
👍 for creating util to execute bun CLI and returning the yarn lock file contents as string. The result can then be passed to `stringifyYarnLockFile` since it should compatible (just check for `# bun` and `version "workspace:..."` strings. We may or may not need to sanitize.

Reply:
Just to confirm, you wanting to create a yarn.lock file here (with the additional bun hash?)

Reply:
We can't create bun hash since we can't generate `bun.lockb` to calculate hash off.

But by having a `yarn.lock` file, when running `bun install` it should use versions of the packages specified in `yarn.lock` to construct new bun lock file. At least that is what I'm hoping for.

Can you verify this?

Basically - if you take a yarn lock file with outdated nested dependencies (one of the fixture ones would work fine), then run bun install and finally bun bun.lockb - the result should be the same versions of packages.

Reply:
That's the best we can do right now I'm afraid

Reply:
We can also do the lockfile generation (as `yarn.lock`) in a follow-up PR. I don't think we need to handle it here.

Reply:
I'm looking at generating the binary file. Happy to update snapshot so this can go in and work on that after? Also love to get involved in adding runtime and bun test. I have plugin that does it for now, but be nice if NX can add native support

Reply:
@jaysoo  If you want to make the yarn.lock a follow up then I think this PR is completed?

Reply:
I reused Yarn in my PR bro. They still had problems with it. The people want Bun support! 

If it builds and runs, then that's a start.

Help us brothers out for taking the time to do this for you.

It was my birthday btw when I built a working version for your project.

Reply:
This is different, this is to handle the manual creation of lock files not the reading of the lock file. Nx team doing a great job BTW, they been so many unstable issues. The chunk of this was taken from my PR back in Sep 11, 2023. But I closed it due to Bun having too many bugs

Reply:
@laynef  you can always use nx-bun to get started for now